dependencies-in-practice.Rmd

```diff
@@ -678,35 +678,24 @@ export(count)
 
 Now imagine that your package implements a method for `count()` for a class you "own" (not `data.frame`).
 A good example is the dbplyr package, which implements `count()` for the `tbl_lazy` class.
-An add-on package that implements an S3 generic for a new class should list the generic-providing package in `Imports`, import the generic into its namespace, and export its S3 method.
-Here's part of dbplyr's `DESCRIPTION` file:
-
-```         
-Imports: 
-    ...,
-    dplyr,
-    ...
-```
 
-In `dbplyr/R/verb-count.R`, we have:
+In this case, `@export` will not work because it assumes the `count()` generic is included or imported in the dbplyr NAMESPACE, so instead we need to use `@exportS3Method`.
+This will use "conditional" method registration, which means the method will only be registered when the dplyr package is loaded.
```

Comment:
I think it would be worth a sentence or two here (or maybe below) to also mention that this also works even if the package is only suggested.

Reply:
Updated with a sentence below (and removes mention of "conditional" method registration for import dependencies)